app/src/main/res/layout/activity_main.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/container"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+
+    <FrameLayout
+        android:id="@+id/fragment_container"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintBottom_toTopOf="@+id/navigation" />
+
+    <!-- add this to fix button shifting and lack of labels once support 28.0.0 or androidx 1.0.0 is available -->
+    <!--         app:labelVisibilityMode="labeled" -->
+    <android.support.design.widget.BottomNavigationView

```

Comment:
Once we update to API 28, we'll be able to show the labels all the time and stop the button resizing/shifting behavior.